core/changedetector/proto_change_detector_service.go

```diff
@@ -0,0 +1,215 @@
+				}
+				for _, newEnumDesc := range newMessageDesc.GetEnumType() {
+					enumName := messageName + "." + newEnumDesc.GetName()
+					if oldEnumDesc := findEnumDescriptorFromMessageDescriptor(oldMessageDesc, newEnumDesc.GetName()); oldEnumDesc != nil {
+						if !proto.Equal(newEnumDesc, oldEnumDesc) {
+							sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+							appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+						}
+					} else {
+						sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+						appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+					}
+				}
+			} else {
+				sce.UpdatedSchemas = append(sce.UpdatedSchemas, messageName)
+				appendImpactedFields(sce, messageName, getImpactedMessageFields(oldMessageDesc, newMessageDesc))
+			}
+		}
+		compareEnumDescriptors(fds, packageEnumMap, sce)
+	}
+}
+func appendImpactedFields(sce *stencilv1beta2.SchemaChangedEvent, key string, impactedFields []string) {
+	if sce.UpdatedFields == nil {
+		sce.UpdatedFields = make(map[string]*stencilv1beta2.ImpactedFields)
+	}
+	if val, ok := sce.UpdatedFields[key]; ok {
+		val.FieldNames = append(val.FieldNames, impactedFields...)
+	} else {
+		sce.UpdatedFields[key] = &stencilv1beta2.ImpactedFields{
+			FieldNames: impactedFields,
+		}
+	}
+}
+
+func compareEnumDescriptors(fds *descriptorpb.FileDescriptorProto, packageEnumMap map[string]map[string]*descriptor.EnumDescriptorProto, sce *stencilv1beta2.SchemaChangedEvent) {
+	for _, newEnumDesc := range fds.GetEnumType() {
+		enumName := fds.GetPackage() + "." + newEnumDesc.GetName()
+		if oldEnumDesc := getEnumDescriptor(packageEnumMap, fds.GetPackage(), newEnumDesc.GetName()); oldEnumDesc != nil {
+			if !proto.Equal(newEnumDesc, oldEnumDesc) {
+				sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+				appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+			}
+		} else {
+			sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+			appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+		}
+	}
+}
+
+func getPackageMessageMap(fileDescriptorSet *descriptor.FileDescriptorSet) map[string]map[string]*descriptor.DescriptorProto {
+	packageMessageMap := make(map[string]map[string]*descriptor.DescriptorProto)
+	for _, fileDescriptor := range fileDescriptorSet.GetFile() {
+		pkgName := fileDescriptor.GetPackage()
+		if _, ok := packageMessageMap[pkgName]; !ok {
+			packageMessageMap[pkgName] = make(map[string]*descriptor.DescriptorProto)
+		}
+		for _, messageDescriptor := range fileDescriptor.GetMessageType() {
+			packageMessageMap[pkgName][messageDescriptor.GetName()] = messageDescriptor
+		}
+	}
+	return packageMessageMap
+}
+
+func getPackageEnumMap(fileDescriptorSet *descriptor.FileDescriptorSet) map[string]map[string]*descriptor.EnumDescriptorProto {
+	packageMessageMap := make(map[string]map[string]*descriptor.EnumDescriptorProto)
```

Comment:
Can you please explain "When using structure like this, you can alias the type to make it more readable and above alias you can document, what it would contain."?
Rest document I have updated